src/Umbraco.Web.UI.Client/src/common/filters/nestedcontent.filter.js

```diff
@@ -3,67 +3,78 @@
 
 // Cache for node names so we don't make a ton of requests
 var ncNodeNameCache = {
-    id: "",
-    keys: {}
+  id: "",
+  keys: {}
 };
 
 angular.module("umbraco.filters").filter("ncNodeName", function (editorState, entityResource) {
 
-    function formatLabel(firstNodeName, totalNodes) {
-        return totalNodes <= 1
-            ? firstNodeName
-            // If there is more than one item selected, append the additional number of items selected to hint that
-            : firstNodeName + " (+" + (totalNodes - 1) + ")";
+  function formatLabel(firstNodeName, totalNodes) {
+    return totalNodes <= 1
+      ? firstNodeName
+      // If there is more than one item selected, append the additional number of items selected to hint that
+      : firstNodeName + " (+" + (totalNodes - 1) + ")";
+  }
+
+  nodeNameFilter.$stateful = true;
+  function nodeNameFilter(input) {
+
+    // Check we have a value at all
+    if (typeof input === 'undefined' || input === "" || input.toString() === "0" || input === null) {
+      return "";
     }
 
-    return function (input) {
+    var currentNode = editorState.getCurrent();
 
-        // Check we have a value at all
-        if (input === "" || input.toString() === "0") {
-            return "";
-        }
+    // Ensure a unique cache per editor instance
+    var key = "ncNodeName_" + currentNode.key;
+    if (ncNodeNameCache.id !== key) {
+      ncNodeNameCache.id = key;
+      ncNodeNameCache.keys = {};
+    }
 
-        var currentNode = editorState.getCurrent();
+    // MNTP values are comma separated IDs. We'll only fetch the first one for the NC header.
+    var ids = input.split(',');
+    var lookupId = ids[0];
+    var serviceInvoked = false;
 
-        // Ensure a unique cache per editor instance
-        var key = "ncNodeName_" + currentNode.key;
-        if (ncNodeNameCache.id !== key) {
-            ncNodeNameCache.id = key;
-            ncNodeNameCache.keys = {};
-        }
+    // See if there is a value in the cache and use that
+    if (ncNodeNameCache.keys[lookupId]) {
+      return formatLabel(ncNodeNameCache.keys[lookupId], ids.length);
+    }
+
+    // No value, so go fetch one
+    // We'll put a temp value in the cache though so we don't
+    // make a load of requests while we wait for a response
+    ncNodeNameCache.keys[lookupId] = "Loading...";
 
-        // MNTP values are comma separated IDs. We'll only fetch the first one for the NC header.
-        var ids = input.split(',');
-        var lookupId = ids[0];
+    // If the service has already been invoked, don't do it again
+    if (serviceInvoked) {
+      return formatLabel(ncNodeNameCache.keys[lookupId], ids.length);
+    }
 
-        // See if there is a value in the cache and use that
-        if (ncNodeNameCache.keys[lookupId]) {
-            return formatLabel(ncNodeNameCache.keys[lookupId], ids.length);
+    serviceInvoked = true;
+
+    var type = lookupId.indexOf("umb://media/") === 0
```

Comment:
@iOvergaard should we use `udiParser.parse()` to get entity type?
https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Web.UI.Client/src/common/services/udiParser.service.js#L24

Reply:
You are right, @bjarnef. We should definitely do that. I guess there are some leftovers in the code base that don't follow the service pattern. I will update this instance even though it doesn't directly relate to this PR.

Reply:
Done